src/convert/metadataConverter.ts

```diff
@@ -49,10 +49,36 @@ export class MetadataConverter {
     components: Iterable<SourceComponent>,
     targetFormat: SfdxFileFormat,
     output: ConvertOutputConfig
+  ): Promise<ConvertResult>;
+
+  /**
+   * Convert metadata components within a `ComponentSet` to another SFDX file format.
+   *
+   * @param componentSet ComponentSet to convert
+   * @param targetFormat Format to convert the component files to
+   * @param output Configuration for outputting the converted files
+   */
+  public async convert(
+    componentSet: ComponentSet,
+    targetFormat: SfdxFileFormat,
+    output: ConvertOutputConfig
+  ): Promise<ConvertResult>;
```

Comment:
This overloads the convert function to accept a ComponentSet or list of SourceComponents.  This is a non-breaking way to support both, and passing the ComponentSet is required if you want to maintain the same sourceApiVersion and apiVersion for the convert.